aot-std-optimizers/src/main/java/io/micronaut/aot/std/sourcegen/Logback14GeneratorHelper.java

```diff
@@ -315,8 +316,13 @@ private boolean maybeGenerateAddOrSet(ImplicitModel model, Model parent, BiFunct
                             return true;
                         } else {
                             try {
-                                Method valueOf = parameterType.getDeclaredMethod("valueOf", String.class);
-                                codeBuilder.addStatement("$L.$L($T.valueOf($S))", parentVarName, method.getName(), ClassName.get(parameterType), model.getBodyText());
+                                if (Charset.class.equals(parameterType)) {
+                                    parameterType.getDeclaredMethod("forName", String.class);
```

Comment:
```suggestion
```

Not necessary since the type is fixed.

Reply:
fixed

Reply:
@yawkat As you see, you're not right. My first fix was right. LogLevel uses valueOf method, forName needs only for Charset

Reply:
If you told me about this line `parameterType.getDeclaredMethod("forName", String.class);` - you're not right too. You need to set argument type - `String.class`, otherwise you will see exception message

Reply:
@yawkat For a long time I could not understand what you meant. Only now everything became clear. So, I just don't see the code that you see.

Look at the screenshot with the changes

![изображение](https://github.com/user-attachments/assets/e4743d75-27fd-4aa1-8ecc-b3092cf8b565)

Reply:
@yawkat  Also, if you are talking about calling a method, then I think that this would be done to get an exception at the optimization stage, i.e. the exception is thrown in the optimizer code. Otherwise, the code will be generated and you will receive an error only at the compilation stage and it will be much more difficult to understand where the error is. So I left the call to this method